File: app/src/release/java/com/example/PerBuildComponentProvider.java
package com.example;

import com.example.api.HttpClientWithPinningProvider;

import okhttp3.OkHttpClient;

class PerBuildComponentProvider {
	private static PerBuildComponentProvider instance;

	static PerBuildComponentProvider getInstance() {
		if (instance == null) {
			instance = new PerBuildComponentProvider();
		}

		return instance;
	}

	OkHttpClient.Builder okHttpBuilder() {
		return HttpClientWithPinningProvider.getClient().newBuilder();
	}
}

File: app/src/release/kotlin/com/example/PerBuildComponentProvider.kt
package com.example

import com.example.api.HttpClientWithPinningProvider

import okhttp3.OkHttpClient

internal class PerBuildComponentProvider {
  fun okHttpBuilder(): OkHttpClient.Builder {
    return HttpClientWithPinningProvider.client.newBuilder()
  }

  companion object {
    private var instance: PerBuildComponentProvider? = null

    fun getInstance(): PerBuildComponentProvider {
      if (instance == null) {
        instance = PerBuildComponentProvider()
      }

      return instance as PerBuildComponentProvider
    }
  }
}
